lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -1872,6 +1872,12 @@ ThreadContext::PushEntryExitRecord(Js::ScriptEntryExitRecord * record)
         Assert(lastRecord->leaveForHost || lastRecord->leaveForAsyncHostOperation);
         lastRecord->hasReentered = true;
         record->next = lastRecord;
+
+        // these are on stack, which grows down. if this condition doesn't hold, then the list somehow got messed up
+        if ((uintptr_t)record > (uintptr_t)lastRecord)
+        {
+            RaiseFailFastException(nullptr, nullptr, 0);
```

Comment:
Why not use ReportFatalException similar to Binary_Inconsistency_fatal_error with a different ErrorReason